### PR TITLE
Adds documentation for voter registration tracking source

### PR DESCRIPTION
**Pull request by aaronschachter**
### What's this PR do?

This pull request fixes the TODO of adding details on the `source` parameter we send along to the Rock The Vote registration site, and how it's used by the Chompy import. It also adds some additional information, in attempt to remove it out of brain and into written words:

* Restores docs for the `VoterRegistrationAction`, removed in #1369. Adds context to why we may see Turbovote links here

* Adds section for a `StartVoterRegistrationForm`

* Adds details about the voting portal's `r` query parameter, and influencer pages

* Specifies the `SitewideBanner` is currently hardcoded to link to the voting portal

### How should this be reviewed?

One of my favorite features of Gitbook is that this can be previewed on the site: https://app.gitbook.com/@dosomething/s/phoenix-documentation/v/docs%2Ftracking-source/development/features/voter-registration#voter-registration-action

👀 📖 

### Any background context you want to provide?

Next up, updating  Chompy docs. Example -- [this section on OVRD](https://github.com/DoSomething/chompy/tree/master/docs/imports#online-drives) can now just link out to these docs.

### Relevant tickets

* [Pivotal #172635133](https://www.pivotaltracker.com/story/show/172635133)
* https://github.com/DoSomething/phoenix-next/pull/2164#discussion_r428714489

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.

### CODE REVIEW

File: docs/development/content-types/voter-registration-action.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+The Voter Registration Action content type renders a visual component which prompts DoSomething.org members to register to vote by linking to an external voter registration partner.
+
+![Voter Registration Action component](../../.gitbook/assets/voter-registration-action-component.png)
+
+## Usage Instructions
+
+When creating a new _Voter Registration Action_ on Contentful, an editor is required to enter a `link` field -- which is the full URL where the member should be directed to upon clicking on the button to start their registration.
+
+- For current entries, this will always be set to a URL in our [vote.dosomething.org Voting Portal]
```
Comment (mendelB) [inline docs/development/content-types/voter-registration-action.md:11]: It looks like a small typo in the link here? 
> [vote.dosomething.org Voting Portal]

File: docs/development/features/sitewide-banner.md [hunk 2]
```diff
@@ -4,7 +4,9 @@
 
 This is a banner feature that is displayed site wide when turned on. In the past, we partnered with a 3rd party service to achieve this (HelloBar), but ran into some unforeseen and hard to troubleshoot bugs. And so, HowdyBar was born!
 
-Use this feature to display important info to our users in a prominent banner across the top of the site. This can be used in conjunction with a `DissmissableElement` or other wrapper components.
+We use this feature to display important info to our users in a prominent banner across the top of the site. It is currently hardcoded to link to our [voter registration portal](/development/features/voter-registration.md#voting-portal).
+
+The `SitewideBanner` can be used in conjunction with a `DissmissableElement` or other wrapper components.
```
Comment (mendelB) [inline docs/development/features/sitewide-banner.md:9]:
```suggestion
The `SitewideBanner` can be used in conjunction with a `DismissableElement` or other wrapper components.
```

File: docs/development/features/voter-registration.md [hunk 3]
```diff
@@ -2,19 +2,75 @@
...
+The [`VoterRegistrationAction` content type](development/content-types/voter-registration-action.md) can be used to display a call to action button that redirects a user to our vote.dosomething.org portal.
+
+A `clicked_voter_registration_action` analytics event is fired when the user clicks on the CTA to visit the voter registration portal.
+
+## Start Voter Registration Form
+
+The `StartVoterRegistrationForm` component displays form fields for email and zip, and redirects a user directly to the RTV registration site upon submitting. It's currently hardcoded on the [OVRD alpha page](#alpha-page) and [Quiz Result page](#quiz-result-page).
 
-## Online Registration Drives
+A `clicked_voter_registration_action` analytics event is fired when the user clicks on the CTA to continue their voter registration on the RTV registration site.
```
Comment (mendelB) [inline docs/development/features/voter-registration.md:71]: Should we explicitly include "when they _submit_ the form" as well? (e.g. via ENTER key. Which does raise the point that the event name might be slightly misleading now, though I suppose we infer this via context 🤷‍♂️ )

**Comment by mendelB** [comment]
> One of my favorite features of Gitbook is that this can be previewed on the site: https://app.gitbook.com/@dosomething/s/phoenix-documentation/v/docs%2Ftracking-source/development/features/voter-registration#voter-registration-action

This is neat! Does it deploy automatically, and accessible via the branch name?

File: docs/development/content-types/voter-registration-action.md (hunk 1 above)
Comment (aaronschachter) [reply docs/development/content-types/voter-registration-action.md:11]: 👀 🦅

File: docs/development/features/voter-registration.md (hunk 3 above)
Comment (aaronschachter) [reply docs/development/features/voter-registration.md]: Updated this. In megagroom, we had landed on keeping the same event name when adding this form. Also had this discussion in [Slack](https://dosomething.slack.com/archives/C03T8SDJG/p1588956298134300?thread_ts=1588955865.133700&cid=C03T8SDJG) that led to including a `contextSource` to include the analytics event:

https://github.com/DoSomething/phoenix-next/blob/04e7db7f99e5afd8c1811f2a484ed73a47039b94/resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaPage.js#L109-L115

**Comment by aaronschachter** [comment]
>This is neat! Does it deploy automatically, and accessible via the branch name?

Only when your branch uses the `docs/*` prefix. Occasionally it doesn't work (you don't see your branch within Gitbook after pushing) but I haven't figured out what the 🐛 is there

File: docs/development/features/voter-registration.md (hunk 3 above)
Comment (mendelB) [reply docs/development/features/voter-registration.md]: I do recall this. I might've failed to realize that this was a form submission as opposed to button click. (My bad, it's outlined in the Pivotal ticket). 

No biggie, I suppose we'll be able to infer the context in any case, and the `contextSource` will help!

**Comment by mendelB** [comment]
> > This is neat! Does it deploy automatically, and accessible via the branch name?
> 
> Only when your branch uses the `docs/*` prefix. Occasionally it doesn't work (you don't see your branch within Gitbook after pushing) but I haven't figured out what the 🐛 is there

Ah this is so helpful! I'll have to remember to do this next time I have a docs PR.

**Pull request merged**